tests/e2e/utils/workspace/TestWorkspaceUtil.ts

```diff
@@ -46,46 +57,217 @@ export class TestWorkspaceUtil implements ITestWorkspaceUtil {
             }
 
             await this.driverHelper.wait(polling);
         }
     }
 
-    public async getIdOfRunningWorkspace(namespace: string): Promise<string> {
-        throw new Error('Method not implemented.');
+    public async getListOfWorkspaceId() {
+        const getAllWorkspacesResponse: rm.IRestResponse<any> = await this.processRequest(RequestType.GET, this.workspaceApiUrl);
+
+        interface IMyObj {
+            id: string;
+            status: string;
+        }
+        let stringified = JSON.stringify(getAllWorkspacesResponse.result);
+        let arrayOfWorkspaces = <IMyObj[]>JSON.parse(stringified);
+        let wsList: Array<string> = [];
+        for (let entry of arrayOfWorkspaces) {
+            wsList.push(entry.id);
+        }
+        return wsList;
+    }
+
+    public async getIdOfRunningWorkspaces(): Promise<Array<string>> {
+        try {
+            const getAllWorkspacesResponse: rm.IRestResponse<any> = await this.processRequest(RequestType.GET, this.workspaceApiUrl);
+
+            interface IMyObj {
+                id: string;
+                status: string;
+            }
+            let stringified = JSON.stringify(getAllWorkspacesResponse.result);
+            let arrayOfWorkspaces = <IMyObj[]>JSON.parse(stringified);
+            let idOfRunningWorkspace: Array<string> = new Array();
+
+            for (let entry of arrayOfWorkspaces) {
+                if (entry.status === 'RUNNING') {
+                    idOfRunningWorkspace.push(entry.id);
+                }
+            }
+
+            return idOfRunningWorkspace;
+        } catch (err) {
+            console.log(`Getting id of running workspaces failed. URL used: ${this.workspaceApiUrl}`);
+            throw err;
+        }
+
     }
 
-    removeWorkspaceById(id: string): void {
+    public async removeWorkspaceById(id: string) {
+        const workspaceIdUrl: string = `${this.workspaceApiUrl}/${id}`;
+        const attempts: number = TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS;
+        const polling: number = TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_POLLING;
+        let stopped: Boolean = false;
+
+        for (let i = 0; i < attempts; i++) {
+
+            const getInfoResponse: rm.IRestResponse<any> = await this.processRequest(RequestType.GET, workspaceIdUrl);
+
+            if (getInfoResponse.result.status === 'STOPPED') {
+                stopped = true;
+                break;
+            }
+            await this.driverHelper.wait(polling);
+        }
+
+        if (stopped) {
+            try {
+                const deleteWorkspaceResponse: rm.IRestResponse<any> = await this.processRequest(RequestType.DELETE, workspaceIdUrl);
+
+                // response code 204: "No Content" expected
+                if (deleteWorkspaceResponse.statusCode !== 204) {
+                    throw new Error(`Can not remove workspace. Code: ${deleteWorkspaceResponse.statusCode} Result: ${deleteWorkspaceResponse.result}`);
```

Comment:
Here and everywhere in the code, it would be great to use specific error type except general.